modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/upload/ImageBlogsUploadFileEntryHandler.java

```diff
@@ -119,6 +126,29 @@ protected FileEntry addFileEntry(
 	@Reference(target = "(resource.name=" + BlogsConstants.RESOURCE_NAME + ")")
 	protected PortletResourcePermission portletResourcePermission;
 
+	private FileEntry _editImageFileEntry(
+			UploadPortletRequest uploadPortletRequest,
+			ThemeDisplay themeDisplay)
+		throws IOException, PortalException {
+
+		long fileEntryId = ParamUtil.getLong(
```

Comment:
@carloslancha The first time I upload and edit the image on the same action, the id is "undefined"

In the previous version we did not edit to the reciently upload image.
But, if the fileEntryId can be passed OR the filename, this can be done

In other case, maybe we must hide the edit option if the id is not present.

Reply:
Updated the pr sending correct information.

Current Item selector is quite a mess, confusing API, receiving different stuff from server after upload, strange data massaging... We (lima) definitelly need to rethink about the whole component, it's too fragile.